pkg/kapp/app/recorded_app.go

```diff
@@ -130,6 +130,43 @@ func (a *RecordedApp) UpdateUsedGVsAndGKs(gvs []schema.GroupVersion, gks []schem
 func (a *RecordedApp) CreateOrUpdate(labels map[string]string) error {
 	defer a.logger.DebugFunc("CreateOrUpdate").Finish()
 
+	app, foundMigratedApp, err := a.find(a.fqName())
+	if err != nil {
+		return err
+	}
+
+	if foundMigratedApp {
+		a.isMigrated = true
+		return a.updateApp(app, labels)
+	}
+
+	app, foundNonMigratedApp, err := a.find(a.name)
+	if err != nil {
+		return err
+	}
+
+	if !foundNonMigratedApp {
+		return a.create(labels)
+	}
+
+	if a.isMigrationEnabled() {
+		return a.migrate(app, labels, a.fqName())
+	}
+	return a.updateApp(app, labels)
```

Comment:
Does this make more sense? The current way is sort of a standard way of doing things in Go, but in this context I felt that it would make more sense to call `create` at the end.
```suggestion
	if foundNonMigratedApp {
	        if a.isMigrationEnabled() {
		       return a.migrate(app, labels, a.fqName())
	        }
	        return a.updateApp(app, labels)
	}
	return a.create(labels)
```

Reply:
I like it !